app/models/compute_resources/foreman/model/libvirt.rb

```diff
@@ -170,10 +170,15 @@ def create_vm(args = { })
       raise e
     end
 
+    def console_password_length(type)
+      # use an 8 character password for vnc (which is the allowed maximum) and 16 for everything else
+      type == 'vnc' ? 8 : 16
+    end
```

Comment:
I think this would make sense in compute_resources.rb and then reuse it in vmware too. The VNC password length is a protocol limitation so having that in a single place feels correct to me.

Reply:
I thought about that when I wrote it, but decided against it. I've added the method to handle the differences between vnc and spice in the libvirt plugin. I think for now it's best to leave the setting of password lengths to the plugins. In the future it would be nice to allow users to set their own password lengths for certain console types. If this is implemented would this make use of a ``console_password_length`` method or would this be passed to the plugins via vm attributes ?